code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,300 @@
+		M.emote("spin")
+
+///This proc listens to the flip signal and throws the mob every third flip
+/datum/reagent/drug/blastoff/proc/on_flip()
+	SIGNAL_HANDLER
+
+	if(!iscarbon(holder.my_atom))
+		return
+	var/mob/living/carbon/dancer = holder.my_atom
+
+	flip_count++
+	if(flip_count >= BLASTOFF_DANCE_MOVES_PER_SUPER_MOVE) //Do a super flip
+		flip_count = 0
+		var/atom/throw_target = get_edge_target_turf(dancer, dancer.dir)
+		dancer.SpinAnimation(12,3)
+		dancer.visible_message(span_notice("[dancer] does an extravagant flip!"), span_nicegreen("You do an extravagant flip!"))
+		dancer.throw_at(throw_target, range = 4, speed = overdosed ? 4 : 1)
+
+///This proc listens to the spin signal and throws the mob every third spin
+/datum/reagent/drug/blastoff/proc/on_spin()
+	SIGNAL_HANDLER
+
+	if(!iscarbon(holder.my_atom))
+		return
+	var/mob/living/carbon/dancer = holder.my_atom
+
+	spin_count++
+	if(spin_count >= BLASTOFF_DANCE_MOVES_PER_SUPER_MOVE) //Do a super flip
+		spin_count = 0
+		dancer.visible_message(span_danger("[dancer] spins around violently!"), span_danger("You spin around violently!"))
+		dancer.spin(30, 2)
+		if(dancer.disgust < 40)
+			dancer.adjust_disgust(10)
+		if(!dancer.pulledby)
+			return
+		var/dancer_turf = get_turf(dancer)
+		var/atom/movable/dance_partner = dancer.pulledby
+		dance_partner.visible_message(span_danger("[dance_partner] tries to hold onto [dancer], but is thrown back!"), span_danger("You try to hold onto [dancer], but you are thrown back!"), null, COMBAT_MESSAGE_RANGE)
+		var/throwtarget = get_edge_target_turf(dancer_turf, get_dir(dancer_turf, get_step_away(dance_partner, dancer_turf)))
+		if(overdosed)
+			dance_partner.throw_at(target = throwtarget, range = 7, speed = 4)
+		else
+			dance_partner.throw_at(target = throwtarget, range = 4, speed = 1) //superspeed
+
+/datum/reagent/drug/saturnx
+	name = "SaturnX"
+	description = "This compound was first discovered during the infancy of cloaking technology and at the time thought to be a promising candidate agent. It was withdrawn for consideration after the researchers discovered a slew of associated safety issues including thought disorders and hepatoxicity."
+	reagent_state = SOLID
+	taste_description = "metallic bitterness"
+	color = "#638b9b"
+	overdose_threshold = 25
+	ph = 10
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/maintenance_drugs = 20)
+
+/datum/reagent/drug/saturnx/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	. = ..()
+	M.adjustOrganLoss(ORGAN_SLOT_LIVER, 0.6 * REM * delta_time)
+
+/datum/reagent/drug/saturnx/on_mob_metabolize(mob/living/L)
+	. = ..()
+	playsound(L, 'sound/chemistry/saturnx_fade.ogg', 30)
+	to_chat(L, span_nicegreen("You feel pins and needles all over your skin as your body suddenly becomes transparent!"))
+	addtimer(CALLBACK(src, .proc/turn_man_invisible, L), 10) //just a quick delay to synch up the sound.
```

Comment:
if you take a sip of saturnx, strap yourself to a stasis bed within 1 second of doing so, and then get your liver removed while on said stasis bed, you can remain invisible indefinitely.

Reply:
I've fixed this i think.